CM7/Core/Src/gatedriver.c

```diff
@@ -81,10 +90,62 @@ int16_t gatedrv_read_dc_current(gatedriver_t* drv)
 }
 
 /* Note: This has to atomically write to ALL PWM registers */
-//TODO: mechanism for PWM synchronization
-int16_t gatedrv_write_pwm(gatedriver_t* drv)
+int16_t gatedrv_write_pwm(gatedriver_t* drv, float dutyCycles[])
 {
```

Comment:
This is fairly important: We should wrap this whole function in a mutex. We should store the mutex in the instance struct and initialize in init, then lock before calling HAL functions and unlock after we are done. There could be a race condition in which we might have another function like `stop_motor` that will try and hard set the PWM in another thread or somethin

Reply:
https://github.com/Northeastern-Electric-Racing/Cerberus/blob/d85e8d368eacff076d41d5849132ded3119a8ed1/Core/Src/mpu.c#L49
Reference on how to use a mutex in Cerberus

Reply:
https://www.keil.com/pack/doc/CMSIS/RTOS2/html/group__CMSIS__RTOS__MutexMgmt.html
Also this

Reply:
Makes sense, adds thread-safety. I'll add the mutex in.